Added "Bargonia" bar sign!

DrAmazing343 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
I made this submission for the Bar Sign contest, and while I didn't make it into any of the higher accolades, I'd still love to see it make it into the game!
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Cargo's always had fun making all sorts of structures in the warehouse, and the fabled Cargo Bar is one of these that seems to be a fan-favorite. I set out to make a sign to be a nice nod to these bars, and for something else for people to put up on the wall to decorate the bars!
![bargoniaupscaledx4](https://github.com/tgstation/tgstation/assets/147954468/8587aadc-a99e-4652-8561-4befac44e294)

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl: Jane
add: Bargonia Bar Sign for Cargo Bar Enjoyers
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DrAmazing343 committed:
Added "Bargonia" to barsigns.dm & barsigns.dmi

DrAmazing343 committed:
Updated neon lighting to be defined as COLOR_WHITE
instead of #ffffff!

BlueMemesauce commented:
It feels more like a bar nation name rather than a cargo reference

san7890 commented on code/game/machinery/barsigns.dm:
```diff
@@ -394,6 +394,12 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/barsign, 32)
 
+/datum/barsign/bargonia
+    name = "Bargonia"
+    icon = "bargonia"
```
```suggestion
    icon_state = "bargonia"
```

your code won't compile otherwise

DrAmazing343 replied:
Committed the suggestion. Thank you!

DrAmazing343 committed:
Update code/game/machinery/barsigns.dm

Co-authored-by: san7890 <the@san7890.com>

DrAmazing343 commented:
> It feels more like a bar nation name rather than a cargo reference

We call Cargo that's "seceded" from the station Cargonia. Bargonia is a reference to the term Cargonia.

deathrobotpunch commented:
> > It feels more like a bar nation name rather than a cargo reference
> 
> We call Cargo that's "seceded" from the station Cargonia. Bargonia is a reference to the term Cargonia.

*Bargo

Loafin34 commented:
why a mothroach? why not a mulebot?

DrAmazing343 commented:
> why a mothroach? why not a mulebot?

Mulebot would be fun, but I really liked the visual of cracking open a wooden crate and mothroaches come in the wooden critter containers!

ZephyrTFA reviewed (changes requested):
We just got a Mothroach themed sign, I'm not inserted in this one also having one at the moment

DrAmazing343 commented:
> We just got a Mothroach themed sign, I'm not inserted in this one also having one at the moment

Is there a level of quality issue with my sign? Any constructive criticism aside from a disagreement in theming? Something concrete that I can improve on?

ZephyrTFA commented:
> > We just got a Mothroach themed sign, I'm not inserted in this one also having one at the moment
> 
> Is there a level of quality issue with my sign? Any constructive criticism aside from a disagreement in theming? Something concrete that I can improve on?

Don't get me wrong. The barsign is great. I just don't want another mothroach one so soon after we just had Maltroach merged. I feel like it detracts from the novelty of the mothroach sign and siphons attention away from it for the wrong reason.
This is in no way a blocking review and just an opinion from me for other maintainers

DrAmazing343 commented:
> > > We just got a Mothroach themed sign, I'm not inserted in this one also having one at the moment
> > 
> > 
> > Is there a level of quality issue with my sign? Any constructive criticism aside from a disagreement in theming? Something concrete that I can improve on?
> 
> Don't get me wrong. The barsign is great. I just don't want another mothroach one so soon after we just had Maltroach merged. I feel like it detracts from the novelty of the mothroach sign and siphons attention away from it for the wrong reason. This is in no way a blocking review and just an opinion from me for other maintainers

Ah, okay, I don't know how all the PR business works, so I was really worried for a moment. I understand your concern, but having spoken to the Maltroach's creator, they quite liked my sign as well! This was my best foot forward to create a nice novel sign for one of my favorite departments/gimmicks; I hope more people don't take it as an attempt to steal thunder from the very very impressive sign that Moff Nyan created.  

If anyone reviewing/looking at the PR has any creative suggestions, I'd love to hear them out as well, since this is my first addition to the codebase and I'd like to have it be the best I can make it if it's gonna make it in at all.

Wallemations commented:
I'd recommend making the letters more 3D, or perhaps even having them be dimmed out before they're lit up rather than simply appearing, more like neon lights rather than just a couple pixels on a purple background?

DrAmazing343 committed:
Reverted change to the .dm because changing "icon"
to "icon_state" caused a compile error. Changed
sign lettering to be more 3D and neon-like,
and accordingly adjusted the lightmask.

DrAmazing343 commented:
> I'd recommend making the letters more 3D, or perhaps even having them be dimmed out before they're lit up rather than simply appearing, more like neon lights rather than just a couple pixels on a purple background?

Right, please check out the new .dmi! I've also updated the PR page's gif to be upscaled to SEE IT, but that's the aseprite raw that doesn't have the .dmi's better timings, so it's mostly just to show off the updated 3D/neon tube'd up sprite.

Wallemations commented:
> > I'd recommend making the letters more 3D, or perhaps even having them be dimmed out before they're lit up rather than simply appearing, more like neon lights rather than just a couple pixels on a purple background?
> 
> Right, please check out the new .dmi! I've also updated the PR page's gif to be upscaled to SEE IT, but that's the aseprite raw that doesn't have the .dmi's better timings, so it's mostly just to show off the updated 3D/neon tube'd up sprite.

Big improvement, though I'll ask if you've tried making the shadow on the top of the letters instead to make them seem like they're popping out from an above view?
No need to resprite the whole thing to check, just the final frame edited to see how it would look should be fine

DrAmazing343 commented:
> > > I'd recommend making the letters more 3D, or perhaps even having them be dimmed out before they're lit up rather than simply appearing, more like neon lights rather than just a couple pixels on a purple background?
> > 
> > 
> > Right, please check out the new .dmi! I've also updated the PR page's gif to be upscaled to SEE IT, but that's the aseprite raw that doesn't have the .dmi's better timings, so it's mostly just to show off the updated 3D/neon tube'd up sprite.
> 
> Big improvement, though I'll ask if you've tried making the shadow on the top of the letters instead to make them seem like they're popping out from an above view? No need to resprite the whole thing to check, just the final frame edited to see how it would look should be fine

Do you mean adding the shadow-shading line to the tops of the lettering as opposed to the left-side? I don't know exactly what area you're talking about here, but I'll try to resprite what I think you're talking about in the morning. If you come back to review in the meantime, a bit of clarification would go a long ways to helping me.

DrAmazing343 commented:
![bargonia top shading test](https://github.com/tgstation/tgstation/assets/147954468/0ed783e4-546f-4f48-bede-f55fd73f0f5a)

Right, trying this one on for size. I think I personally liked the side-lighting better, but if this is what you meant to try and you think it looks better, I'll redo the .dmi/lightmask for it!

Wallemations commented:
> ![bargonia top shading test](https://github.com/tgstation/tgstation/assets/147954468/0ed783e4-546f-4f48-bede-f55fd73f0f5a)
> 
> 
> 
> Right, trying this one on for size. I think I personally liked the side-lighting better, but if this is what you meant to try and you think it looks better, I'll redo the .dmi/lightmask for it!

I think I personally prefer this one, yea
Remember to add the darker colors for the inside of the letters too

DrAmazing343 committed:
Changed barsign and lightmap to include
appropriate shading changes. Fixed
a piece of dropped lighting in the
lightmap.

DrAmazing343 commented:
> I think I personally prefer this one, yea Remember to add the darker colors for the inside of the letters too

Adjusted and committed! Any other issues I need to work out?

Wallemations commented:
the merge conflicts, mostly.
the influx of bar signs means that every time I merge one every other barsign PR has to adjust for it

DrAmazing343 committed:
Merge branch 'master' of https://github.com/DrAmazing343/tgstation into Bargonia

DrAmazing343 committed:
image.png

DrAmazing343 committed:
git pull upstream/master

DrAmazing343 committed:
Hopefully fixed all merge errors!

DrAmazing343 committed:
Merge pull request #1 from DrAmazing343/bargonia

Bargonia

DrAmazing343 committed:
Merge branch 'master' into Bargonia

DrAmazing343 commented:
Ignore most of the fuckery above, I'm a novice and it shows. Everything SHOULD be clear for merging, unless I've missed or messed up something?

DrAmazing343 committed:
Merge branch 'master' into Bargonia

Wallemations reviewed (approved):
since there's a lot goin on here, once that thing ghommie suggested gets fixed if any other maints wanna merge it before I do go ahead
Otherwise I'll check in again later and merge it myself

DrAmazing343 committed:
Update code/game/machinery/barsigns.dm

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>

DrAmazing343 committed:
Merge branch 'master' into Bargonia

Ghommie committed:
Update barsigns.dm
